storage/oasis/nodeapi/history/history.go

```diff
@@ -0,0 +1,293 @@
+		dialOpts := []grpc.DialOption{grpc.WithTransportCredentials(creds)}
+		grpcConn, err := cmnGrpc.Dial(nodeConfig.RPC, dialOpts...)
+		if err != nil {
+			return nil, fmt.Errorf("oasis-core common grpc Dial: %w", err)
+		}
+		return wrap(grpcConn), nil
+	}
+}
+
+var DefaultChains = map[string]*History{
+	"mainnet": {
+		Records: []*Record{
+			{
+				// https://github.com/oasisprotocol/mainnet-artifacts/releases/tag/2022-04-11
+				ArchiveName:   "damask",
+				ConstructAPI:  BindConstructorCurrent(damask.NewDamaskConsensusApiLiteErased),
+				GenesisHeight: 8048956,
+				ChainContext:  "b11b369e0da5bb230b220127f5e7b242d385ef8c6f54906243f30af63c815535",
+			},
+			{
+				// https://github.com/oasisprotocol/mainnet-artifacts/releases/tag/2021-04-28
+				ArchiveName:   "cobalt",
+				ConstructAPI:  BindConstructorGRPC(cobalt.NewCobaltConsensusApiLiteErased),
+				GenesisHeight: 3027601,
+				ChainContext:  "53852332637bacb61b91b6411ab4095168ba02a50be4c3f82448438826f23898",
+			},
+			{
+				// https://github.com/oasisprotocol/mainnet-artifacts/releases/tag/2020-11-18
+				ArchiveName:   "mainnet",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 702000,
+				ChainContext:  "a4dc2c4537992d6d2908c9779927ccfee105830250d903fd1abdfaf42cb45631",
+			},
+			{
+				// https://github.com/oasisprotocol/mainnet-artifacts/releases/tag/2020-10-01
+				ArchiveName:   "beta",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 1,
+				ChainContext:  "a245619497e580dd3bc1aa3256c07f68b8dcc13f92da115eadc3b231b083d3c4",
+			},
+		},
+	},
+	"testnet": {
+		Records: []*Record{
+			// TODO: coalesce compatible records
+			// TODO: enter compatible API
+			// TODO: fill in chain context
+			{
+				// https://github.com/oasisprotocol/testnet-artifacts/releases/tag/2022-03-03
+				ArchiveName:   "2022-03-03",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 8535081,
+			},
+			{
+				// https://github.com/oasisprotocol/testnet-artifacts/releases/tag/2021-04-13
+				ArchiveName:   "2021-04-13",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 3398334,
+			},
+			{
+				// https://github.com/oasisprotocol/testnet-artifacts/releases/tag/2021-03-24
+				ArchiveName:   "2021-03-24",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 3076800,
+			},
+			{
+				// https://github.com/oasisprotocol/testnet-artifacts/releases/tag/2021-02-03
+				ArchiveName:   "2021-02-03",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 2284801,
+			},
+			{
+				// https://github.com/oasisprotocol/testnet-artifacts/releases/tag/2020-11-04
+				ArchiveName:   "2020-11-04",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 811055,
+			},
+			{
+				// https://github.com/oasisprotocol/testnet-artifacts/releases/tag/2020-09-15
+				ArchiveName:   "2020-09-15",
+				ConstructAPI:  NotImplemented,
+				GenesisHeight: 1,
+			},
+		},
+	},
+}
+
+type HistoryConsensusApiLite struct {
+	History *History
+	// APIs. Keys are archive names.
```

Comment:
```suggestion
	// APIs. Keys are lowercase archive names (cobalt, damask, ...).
```

I'm worried about naming. Not that I have any great solutions. But cobalt and damask are chains, aren't they. They each have their _chain_ ID (oasis-2, oasis-3), their _chain_ context. So maybe instead, mainnet vs testnet are ... megachains? metachains? I'm open to suggestions.

Quite possibly, we should just leave names as they are. Maybe there's no hope in hell of getting the naming straight; "chain" seems to be everything from "mainnet" to "damask" to "emerald", not just here but in oasis-core land too. It's partly for that reason that I like peppering chain-related docs with little parenthesized examples to clarify, like in this suggested diff.

Reply:
> I'm open to suggestions.

continuity. or continuum



> cobalt and damask are chains

:face_with_spiral_eyes: also been using the term 'network' due to that one SDK struct

internally I've been calling them 'beta,' 'mainnet-a,' 'mainnet-b,' 'mainnet-c'

Reply:
I kinda like continuum :)
But it's also a rare enough concept that I'd probably still want to mention it as "continuum (e.g. mainnet/testnet/localnet)" in most docstrings.

Reply:
introduce the term separately (: